### PR TITLE
Feature/proj 150

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
This PR introduces the Carousel component, designed to handle image display and browsing in a sleek slider interface. The component provides full integration for adding, scrolling through, and managing multiple images, with customized navigation arrows for a seamless user experience.

### Key Features of the Carousel Component:
**Image Display**: Handles both single and multiple images. If provided with a single image, it automatically adapts and displays it. When multiple images are available, it enables navigation through them using arrows.
Navigation Arrows: Customizable previous and next arrows allow users to scroll through the images. Arrows are hidden if there's only one image.
Supports adding new images through a file input, allowing users to select additional photos. The images are automatically displayed in the carousel after selection.
Responsive Design: The carousel is responsive, adapting its height based on the image content. This ensures that images are displayed optimally regardless of size.

### Usage in ImagePostModal:
In the ImagePostModal component, the Carousel is integrated to display images that are part of a post. The user can browse through the post's images using the slider. If the user enters the cropping step (step === 'cropping'), a file input is provided for adding new images, which are immediately appended to the carousel.

### Usage in CreatePostModal:
In the CreatePostModal component, the Carousel is used to display any images that the user uploads when creating a new post. It supports adding images via the 'Select from Computer' button, which updates the carousel in real-time. The carousel ensures that users can preview all selected images before finalizing their post.

### Key Changes:
Added Carousel component with react-slick slider integration.
Implemented custom next/previous arrow components to handle index updates.
Integrated the file input for adding new images, allowing users to dynamically modify the carousel content.
Added support for the cropping step, enabling additional image upload functionality in the ImagePostModal.

**In the ImagePostModal:**
Open a post with multiple images.
Scroll through the images using the navigation arrows.
Enter the cropping step and add new images via the file input. Ensure the new images appear in the carousel.

**In the CreatePostModal:**
Start creating a post and upload multiple images.
Confirm that the images are displayed correctly in the carousel.
Add more images and verify that the carousel updates to include them.

This PR enhances the user experience by streamlining the image browsing and selection process in both the ImagePostModal and CreatePostModal components.
Cropping

### CODE REVIEW

File: src/features/posts/ui/imagePostModal/CreatePostModal.tsx [hunk 1]
```diff
@@ -3,10 +3,9 @@ import { useEffect, useRef, useState } from 'react'
 import { Avatar, Post } from '@/features'
 import { ArrowIosBackOutline } from '@/shared/assets'
 import { useChangeTitle } from '@/shared/utils/hooks/useChangeTitle'
```
Comment (NoName-ForTeam) [inline src/features/posts/ui/imagePostModal/CreatePostModal.tsx]: @/shared/utils

File: src/features/posts/ui/imagePostModal/imagePostModal.tsx [hunk 2]
```diff
@@ -10,18 +10,19 @@ import {
 import { Close, CloseOutline, Edit2 } from '@/shared/assets'
 import { ProfileAvatar } from '@/shared/ui'
 import { useChangeTitle } from '@/shared/utils/hooks/useChangeTitle'
```
Comment (NoName-ForTeam) [inline src/features/posts/ui/imagePostModal/imagePostModal.tsx]: @/shared/utils

File: src/widgets/slider/slider.module.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
+.dotsItem {
+  cursor: pointer;
+
+  width: 8px;
+  height: 8px;
+  margin: -30px 5px;
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.module.scss]: отрицательный маржин в режиме статического позиционирования явный костыль, не комильфо)

File: src/widgets/slider/slider.module.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.prevArrow {
+  cursor: pointer;
+
+  position: absolute;
+  z-index: 1;
+  top: 50%;
+  left: 0;
+  transform: translate(0, -50%);
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.module.scss]: можно вынести в миксин, поскольку есть посторение ниже

File: src/widgets/slider/slider.tsx [hunk 5]
```diff
@@ -0,0 +1,216 @@
+import { ChangeEvent, useState } from 'react'
+import Slider from 'react-slick'
+
+import { Step } from '@/features'
+import { ArrowIosBackOutline, ArrowIosForwardOutline, ImageOutline } from '@/shared/assets'
+import { useModalAddPhoto } from '@/widgets/modals/ui/modalAddPhoto/useModalAddPhoto'
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.tsx]: @/widgets

File: src/widgets/slider/slider.tsx [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    setImage,
+  })
+
+  const classNames = {
+    dotsItem: styles.dotsItem,
+    dotsItemActive: styles.dotsItemActive,
+    icon: styles.icon,
+    selectedImage: styles.selectedImage,
+    slider: styles.slider,
+  }
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.tsx]: as const

File: src/widgets/slider/slider.tsx [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+  /**
+   * Handles the addition of a new image to the carousel.
+   * Updates the images array with the newly added image(s).
+   */
+  const handleImageAddition = (newImage: string | string[]) => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    setImage(prevImages => {
+      const updatedImages = Array.isArray(prevImages) ? prevImages : [prevImages]
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.tsx:94]: второй раз - можно вынести в utils как функцию, которая проверяет и возвращает массив

File: src/widgets/slider/slider.tsx [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+  callbackFunction,
+  indexArrow,
+  onClick,
+  photosLength,
+  setIndexArrow,
+}: ArrowsProps) => {
+  if (
+    indexArrow === (photosLength ? photosLength - 1 : 0) ||
+    (photosLength ? photosLength : 0) <= 1
+  ) {
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.tsx:163]: хоть убей, но логику не догоняю))

File: src/widgets/slider/slider.tsx [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+   */
+  const handleClick = () => {
+    setIndexArrow(indexArrow + 1)
+    if (callbackFunction) {
+      callbackFunction()
+    }
+    if (onClick) {
+      onClick()
+    }
+  }
```
Comment (NoName-ForTeam) [inline src/widgets/slider/slider.tsx:178]: это из карусели?

File: src/features/posts/ui/imagePostModal/CreatePostModal.tsx (hunk 1 above)
Comment (EvgeniyaYakunina) [reply src/features/posts/ui/imagePostModal/CreatePostModal.tsx]: поправила

File: src/features/posts/ui/imagePostModal/imagePostModal.tsx (hunk 2 above)
Comment (EvgeniyaYakunina) [reply src/features/posts/ui/imagePostModal/imagePostModal.tsx]: поправила

File: src/widgets/slider/slider.tsx (hunk 5 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx:6]: этот импорт сократить не получается

File: src/widgets/slider/slider.tsx (hunk 6 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx]: поправила

File: src/widgets/slider/slider.tsx (hunk 9 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx:178]: да, это логика для правильного функционирования стрелок

File: src/widgets/slider/slider.tsx (hunk 8 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx:163]: Не понимаешь, а она работает)))

File: src/widgets/slider/slider.tsx (hunk 7 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx:94]: учитывая, что у меня тут валится ошибка, я бы вообще хотела отсюда убрать всю эту логику. Пока думаю над решением.

File: src/widgets/slider/slider.module.scss (hunk 4 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.module.scss]: поправила

File: src/widgets/slider/slider.module.scss (hunk 3 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.module.scss]: поправила

**Review by Lelik7777 — APPROVED** [review]

File: src/widgets/slider/slider.tsx (hunk 7 above)
Comment (NoName-ForTeam) [reply src/widgets/slider/slider.tsx:94]: ок

File: src/widgets/slider/slider.tsx (hunk 5 above)
Comment (NoName-ForTeam) [reply src/widgets/slider/slider.tsx]: импортируй хук в индексе, тогда заработает

File: src/widgets/slider/slider.tsx (hunk 8 above)
Comment (NoName-ForTeam) [reply src/widgets/slider/slider.tsx:163]: магия)

File: src/widgets/slider/slider.tsx (hunk 9 above)
Comment (NoName-ForTeam) [reply src/widgets/slider/slider.tsx:178]: твоя логика или из либы карусели?

File: src/widgets/slider/slider.tsx (hunk 5 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx]: поправила

File: src/widgets/slider/slider.tsx (hunk 9 above)
Comment (EvgeniyaYakunina) [reply src/widgets/slider/slider.tsx:178]: Моя. Там танцы с бубном из-за встроенных стрелок. Переопределить им стили не смогла, поэтому запихнула кастомные с нужными стилями.

**Comment by EvgeniyaYakunina** [comment]
run build production

**Comment by Lelik7777** [comment]
run build production

**Pull request merged**